JavaScript/tests/syntax_test_js.js

```diff
@@ -218,10 +218,10 @@ let x = import.meta;
 // method body, but we include it here to ensure that highlighting is not
 // broken as the user is typing
 let a = { otherIdentifier, foo(), baz: 1 }
-//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.object-literal
+//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.mapping
```

Comment:
This wouldn't also satisfy `meta.mapping - meta.mapping meta.mapping`, would it? Because I'm not seeing a `clear_scopes` in this diff, especially not for `object-literal-meta-key`.

Reply:
Good point. One issue is that there's an extra meta scope around the key and value when the value is a function:

```js
+{
    funcKey: function() {
//  ^^^^^^^^^^^^^^^^^^^ meta.mapping meta.function.declaration
//  ^^^^^^^ meta.mapping meta.function.declaration meta.mapping.key entity.name.function
    },
}
```

Wrapping the whole thing like that in `meta.function` is admittedly dubious, but it's used for the symbol list. Do you have any ideas on how to address this?

Reply:
@FichteFoll Any thoughts on this?

Reply:
I think this is reasonable for now. The pattern of naming functions via object key is so pervasive that we definitely wouldn't want to lose that functionality.

Reply:
Well, you could pop two scopes for the mapping key and re-push `meta.mapping.key meta.function entity.name` (and do the same for the value), but unfortunately scope mangling like this can get extremly annoying when a `set` is involved.

There is a solution to this, but having `meta.mapping` is *already* an improvement over `meta.object-literal`.